process/block/shardblock.go

```diff
@@ -895,7 +892,11 @@ func (sp *shardProcessor) CommitBlock(
 	}
 
 	sp.blockChain.SetCurrentBlockHeaderHash(headerHash)
-	sp.indexBlockIfNeeded(bodyHandler, headerHash, headerHandler, lastBlockHeader)
+	err = sp.indexBlockIfNeeded(bodyHandler, headerHash, headerHandler, lastBlockHeader)
```

Comment:
The same comment here like in metablock.go. By moving this call above L803, if it is possible, could save the node from a blocking situation.